autorun.brs

```diff
@@ -298,23 +297,39 @@ Sub EnterEventLoop()
       else if timerData = "checkUpdate" then
         DebugLog("BS: Checking for update")
         versionRequest = CreateObject("roUrlTransfer")
+        versionRequestPort = CreateObject("roMessagePort")
         versionRequest.SetUrl("https://raw.githubusercontent.com/OutOfAxis/fugo-player-bs/main/latest.txt")
-        latestVersion = versionRequest.GetToString().Trim()
-        DebugLog("BS: Latest version: " + latestVersion)
-        if gaa.version = latestVersion then
-          DebugLog("BS: Already up to date")
-        else
-          DebugLog("BS: Retrieving latest autorun.brs")
-          scriptRequest = CreateObject("roUrlTransfer")
-          scriptRequest.SetUrl("https://raw.githubusercontent.com/OutOfAxis/fugo-player-bs/main/autorun.brs")
-          responseCode = scriptRequest.GetToFile("autorun.tmp")
-          DebugLog("BS: Response code = " + stri(responseCode))
-          if responseCode = 200 then
-            DebugLog("BS: Performing update")
-            MoveFile("autorun.brs", "autorun.brs~")
-            MoveFile("autorun.tmp", "autorun.brs")
-            RebootSystem()
+        versionRequest.SetPort(versionRequestPort)
+        if versionRequest.AsyncGetToString() then
+          event = wait(5000, versionRequestPort)
```

Comment:
waiting for 5s for the request to be finished? is this how it's done in Bright Script?

Reply:
for how long it should wait?

Reply:
no, just a weird syntax